### PR TITLE
api: adjust the api structure

### PR DESCRIPTION
**Pull request by zhl003**
### What type of PR is this?

/enhancement
### Which issue(s) this PR fixes?

Fixes #

### What this PR does?

Summary:

### CODE REVIEW

File: api/v1beta1/mysqlcluster_types.go [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+
+	// Network port of the MySQL server to follow via binlog replication.
+	// +optional
+	// +kubebuilder:validation:Minimum=1024
+	Port *int32 `json:"port,omitempty"`
+}
+
+type ServiceSpec struct {
+	// +optional
+	Metadata *Metadata `json:"metadata,omitempty"`
```
Comment (acekingke) [inline api/v1beta1/mysqlcluster_types.go:408]: Maybe do not need it.

File: api/v1beta1/mysqlcluster_types.go [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+
+	// More info: https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types
+	//
+	// +optional
+	// +kubebuilder:default=ClusterIP
+	// +kubebuilder:validation:Enum={ClusterIP,NodePort,LoadBalancer}
+	Type string `json:"type"`
+}
+
+type Metadata struct {
```
Comment (acekingke) [inline api/v1beta1/mysqlcluster_types.go:425]: Maybe do not need it.

File: api/v1beta1/mysqlcluster_types.go [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+	// Whether or not the MySQL cluster should be read-only. When this is
+	// true, the cluster will be read-only. When this is false, the cluster will
+	// run as writable.
+	// +optional
+	// +kubebuilder:default=false
+	Enabled bool `json:"enabled"`
+
+	// The name of the MySQL cluster to follow for binlog.
+	// +optional
+	ClusterName string `json:"repoName,omitempty"`
```
Comment (acekingke) [inline api/v1beta1/mysqlcluster_types.go:394]: Why Call repoName?

File: api/v1beta1/mysqlcluster_types.go [hunk 4]
```diff
@@ -0,0 +1,435 @@
...
+
+	// The compute resource requirements.
+	// +optional
+	// +kubebuilder:default:={limits: {cpu: "100m", memory: "256Mi"}, requests: {cpu: "50m", memory: "128Mi"}}
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+}
+
+type MonitoringSpec struct {
+	// +optional
+	Exporter *ExporterSpec `json:"exporter,omitempty"`
```
Comment (acekingke) [inline api/v1beta1/mysqlcluster_types.go:346]: No need to wrap an extra level.

Comment (acekingke) [reply api/v1beta1/mysqlcluster_types.go:346]: if you want to extend ，use the configmap better.

**Review by acekingke — APPROVED** [review]
ok

**Comment by zhl003** [comment]
Change PR to v3.0.0 branch

**Pull request closed without merging**